### PR TITLE
Foreign currency

### PR DESCRIPTION
**Pull request by Sha-yol**
Hi,
Added functionality to handle transactions in foreign currencies. This solves #10.
For each transaction, I check what is the source account by GETing all asset accounts and finding the one with the matching name. Then it compares its currency code with the expense currency code - if there's a mismatch, my share amount is set as the foreign amount of the transaction, with the SW currency code, and the amount (in the native currency) is set to 0.1. A tag (determined byt a field in the `.env` file) is added to the transaction for later, manual revision.

Current issue: FF is queried for the account list for each SW expense. This doubles the number of requests going to FF for each expense. This could be made more efficient with a global variable holding the account list. This would require only minimal changes to the code as it is. I opted not to add globals to the program since I don't like them - if we find this is an issue, I'm in favor of putting the syncher in a class and passing data that way.

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -348,18 +349,36 @@ def getExpenseTransactionBody(exp: Expense, myshare: ExpenseUser, data: list[str
...
+    if getAccountCurrencyCode(source) != exp.getCurrencyCode():
+        newTxn["foreign_currency_code"] = exp.getCurrencyCode()
+        newTxn["foreign_amount"] = myshare.getOwedShare()
+        newTxn["amount"] = 0.1
+        newTxn["tags"] = [conf["FOREIGN_CURRENCY_TAG"]] 
     print(
         f"Processing {category} {formatExpense(exp, myshare)} from {source} to {dest}")
     return newTxn
 
+def getAccountCurrencyCode(account_name: str) -> str:
```
Comment (adyanth) [inline main.py:368]: You could write a specific memoized version of this function to cache the account list and use it. Something like so:

```python
from functools import wraps

def getAccounts(account_type: str="asset") -> list:
    # Call API here for account_type
    # return callApi("accounts/", method="GET", params={"type": account_type}).json()['data']
    print("Calling API")
    return [
        {
            "attributes": {
                "name": "test1",
                "currency_code": "CD1",
            }
        },
        {
            "attributes": {
                "name": "test2",
                "currency_code": "CD2",
            }
        },
    ]


def cache_account_currency(function):
    account_name_currency = dict(
        map(
            lambda x: (x["attributes"]["name"], x["attributes"]["currency_code"]),
            getAccounts("asset"),
        )
    )

    @wraps(function)
    def cached(account_name: str) -> str:
        # Will throw if account not found, you could update the map if the account might have been recently created. Unlikely.
        return account_name_currency[account_name]

    return cached


@cache_account_currency
def getAccountCurrencyCode(account_name: str) -> str:
    raise Exception("Will not be called")

print(getAccountCurrencyCode("test1"))
print(getAccountCurrencyCode("test2"))
```

File: .env.example [hunk 2]
```diff
@@ -6,3 +6,4 @@ FIREFLY_DEFAULT_TRXFER_ACCOUNT=Charles Schwab
 FIREFLY_DEFAULT_CATEGORY=Groceries
 FIREFLY_DRY_RUN=true
 SPLITWISE_DAYS=1
+FOREIGN_CURRENCY_TAG=fixme/foreign-currency
```
Comment (adyanth) [inline .env.example:9]: nit: rename to FOREIGN_CURRENCY_TOFIX_TAG

File: main.py (hunk 1 above)
Comment (Sha-yol) [reply main.py:390]: Nice, didn't know about memoization. Implemented in a commit.

**Comment by Sha-yol** [comment]
Implemented your suggested changes, how about now?

**Review by adyanth — APPROVED** [review]

**Comment by adyanth** [comment]
Fixes #10

**Pull request merged**